qa/L0_lifecycle/lifecycle_test.py

```diff
@@ -2237,6 +2237,144 @@ def test_config_override(self):
+                except InferenceServerException as ex:
+                    # [FIXME] Improve error reporting to mention missing config
+                    self.assertIn(
+                        "failed to load '{}', failed to poll from model repository"
+                        .format(model_name), ex.message())
+
+                # Sanity check on previous loaded version is still available
+                self.assertFalse(triton_client.is_model_ready(model_name, "1"))
+                self.assertFalse(triton_client.is_model_ready(model_name, "2"))
+                self.assertTrue(triton_client.is_model_ready(model_name, "3"))
+
+                self._infer_success_models([
+                    base[0],
+                ], (3,), model_shape)
+
+                # Request to load the model with override file and config in
+                # a different name
+                try:
+                    triton_client.load_model(
+                        override_model_name,
+                        config="""{{"backend":"{backend}" }}""".format(
+                            backend=base[1]),
+                        encoded_files={"file:1/model.onnx": encoded_file})
+                except Exception as ex:
+                    self.assertTrue(False, "unexpected error {}".format(ex))
+
+                # Sanity check on previous loaded version is still available
+                self.assertFalse(triton_client.is_model_ready(model_name, "1"))
+                self.assertFalse(triton_client.is_model_ready(model_name, "2"))
+                self.assertTrue(triton_client.is_model_ready(model_name, "3"))
+                self._infer_success_models([
+                    base[0],
+                ], (3,), model_shape)
+
+                # New override model should be available
+                self.assertTrue(
+                    triton_client.is_model_ready(override_model_name, "1"))
+                self.assertFalse(
+                    triton_client.is_model_ready(override_model_name, "2"))
+                self.assertFalse(
+                    triton_client.is_model_ready(override_model_name, "3"))
+                self._infer_success_models([
+                    override_base,
+                ], (1,),
+                                           model_shape,
+                                           swap=True)
+
+                # Request to load the model with override file and config in
+                # original name
+                try:
+                    triton_client.load_model(
+                        model_name,
+                        config="""{{"backend":"{backend}" }}""".format(
+                            backend=base[1]),
+                        encoded_files={"file:1/model.onnx": encoded_file})
+                except Exception as ex:
+                    self.assertTrue(False, "unexpected error {}".format(ex))
+
+                # Sanity check on previous loaded version is still available
```

Comment:
I think each of these comments may need to be differentiated. Like here, it looks like you're expecting the load to be successful, which is why 1 is available, right? You're not just checking that the previous versions are still available. Also, is version 3 no longer available because we only allow one version at a time? If so, I wonder if it's necessary to check all of them. If you do so for completeness, you may want to find a way to make it clear what you're specifically checking (since the code ends up being broader, comments may be more important).

Reply:
Yes, the comment should be updated. The reason why version 3 is unavailable is because the override model directory (a new and different model directory) will be used, where it only has version 1.